client/pages/verify.js

```diff
@@ -0,0 +1,218 @@
+/* eslint-disable no-console */
+// Module imports
+import React from 'react'
+// import PropTypes from 'prop-types'
+
+// Component imports
+import { actions, connect, actionStatus } from '../store'
+import { Router, Link } from '../routes'
+import PageWrapper from '../components/PageWrapper'
+import httpStatus from '../helpers/httpStatus'
+import PasswordField from '../components/PasswordField'
+import { passwordPattern } from '../data/RegExpr'
+
+
+// Component Constants
+
+@connect
+class Verify extends React.Component {
+  /***************************************************************************\
+    Class Properties
+  \***************************************************************************/
+
+  state = {
+    password: '',
+    submitted: false,
+    submitting: false,
+    validating: false,
+  }
+
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+  _handleSubmit = async (event) => {
+    const {
+      password,
+      token,
+    } = this.state
+
+    event.preventDefault()
+
+    this.setState({ submitting: true })
+
+    const resetResponse = await this.props.resetPassword({
+      password,
+      token,
+    })
+
+    if (resetResponse.status === actionStatus.SUCCESS) {
+      this.setState({
+        submitted: true,
+        submitting: false,
+      })
+      Router.pushRoute('home', { authenticate: true })
+    }
+  }
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  static async getInitialProps ({ query, store, res }) {
+    // eslint-disable-next-line id-length
+    const { type, t: token } = query
+    let destination = null
+    let response = null
+
+    switch (type) {
+      case 'reset':
+        console.log('Reset Redirect', token)
+        response = await actions.verifyResetToken(token)(store.dispatch)
+        if (response.status === actionStatus.SUCCESS) {
```

Comment:
you could check for status above the switch statement instead and skip it entirely if the status is not a success.

Reply:
I'm not sure how this would work as they all have different tokens.

Reply:
Ah nevermind I was reading the code wrong :P